miden-lib/asm/kernels/transaction/api.masm

```diff
@@ -893,29 +973,39 @@ end
 #! The input block height delta is added to the reference block in order to output an upper limit
 #! up until which the transaction will be considered valid (not expired).
 #!
-#! Inputs: [block_height_delta, ...]
-#! Output: [...]
+#! Inputs: [block_height_delta, pad(15)]
+#! Output: [pad(16)]
 #!
 #! Where:
 #! - block_height_delta is the desired expiration time delta (1 to 0xFFFF).
+#!
+#! Annotations: dynexec
 export.update_expiration_block_num
     exec.tx::update_expiration_block_num
+    # => [pad(16)]
 end
 
 #! Gets the transaction expiration delta.
 #!
-#! Inputs: [...]
-#! Output: [block_height_delta, ...]
+#! Inputs: [pad(16)]
+#! Output: [block_height_delta, pad(15)]
 #!
 #! Where:
 #! - block_height_delta is the stored expiration time delta (1 to 0xFFFF).
+#!
+#! Annotations: dynexec
 export.get_expiration_delta
     exec.tx::get_expiration_delta
+    # => [block_height_delta, pad(16)]
+
+    # truncate the stack
+    swap drop
+    # => [block_height_delta, pad(15)]
```

Comment:
This additional element on the stack was truncated at the end of the `exec_kernel_proc` procedure while executing the `exec.sys::truncate_stack`.

By the way, as far as I can see, we can remove this `truncate_stack` invocation, since all our kernel procedures should leave exactly 16 elements on the stack (otherwise it is a bug).

Reply:
Yeah - let's remove it.